NitroxClient/GameLogic/Bases/Spawning/BasePiece/BaseWaterParkProcessor.cs

```diff
@@ -45,5 +54,15 @@ protected override void SpawnPostProcess(Base latestBase, Int3 latestCell, GameO
                 Log.Debug($"BaseRoomWaterParkProcessor: Created new Waterpark {newWaterparkId} and Planter {newPlanterId}");
             }
         }
+
+        private IEnumerator DelayModuleDetection(Base latestBase, Int3 latestCell, GameObject finishedPiece)
+        {
+            if (!finishedPiece)
+            {
+                yield break;
+            }
+            yield return new WaitForSeconds(0.1f);
+            SpawnPostProcess(latestBase, latestCell, finishedPiece);
+        }
```

Comment:
This should probably moved to base class since this piece code is duplicated in multiple processor